packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -812,7 +812,9 @@ export class DeltaManager
         this.stopSequenceNumberUpdate();
 
         // This raises "disconnect" event
-        this.disconnectFromDeltaStream(error !== undefined ? `${error.message}` : "Container closed");
+        if (this.connection !== undefined) {
```

Comment:
Why move this check outside of `disconnectFromDeltaStream`?  Seems like this makes it easier to call it incorrectly?

Reply:
I can move it in. I change it mostly because old design assumed reentrancy (i.e. event handlers being called on connection that is either not current or in process of being closed). This PR removes any reentrancy (by unregistering handlers on close) and asserts on reentrancy. So any handler calling disconnectFromDeltaStream() on a connection that is not valid/current is violation.

That said, call calls to disconnectFromDeltaStream() from handlers comes through reconnectOnError() that is already checks for reentrancy, so it does not really matter what disconnectFromDeltaStream() does.

Reply:
I'd prob prefer it inside, or else at least add a note in the method doc that a precondition for calling is being in the connected state.  Calling disconnect when in a disconnected state feels like it should be a safe no-op.